File: NO1/untitled.java
import java.util.List;
import java.util.ArrayList;
import java.util.Collections;
// IMPORT LIBRARY PACKAGES NEEDED BY YOUR PROGRAM
// SOME CLASSES WITHIN A PACKAGE MAY BE RESTRICTED
// DEFINE ANY CLASS AND METHOD NEEDED
// CLASS BEGINS, THIS CLASS IS REQUIRED
public class Solution
{ 
	// METHOD SIGNATURE BEGINS, THIS METHOD IS REQUIRED
    public List<String> reorderLines(int logFileSize, List<String> logfile) 
	{
		List<String> list = new ArrayList<String>();
		
		if (logFileSize != logfile.size())
		    return list;
		    
		List<String> digitList = new ArrayList<String>();
		List<String> alphaList = new ArrayList<String>();
		
		for (int i = 0; i < logfile.size(); i++) {
		    String parts[] = logfile.get(i).split("\\s+", 2);
		    if (Character.isDigit(parts[1].charAt(0)))
		        digitList.add(logfile.get(i));
		    else
		        alphaList.add(logfile.get(i));
		}
		
		// sort alphaList with lexicographically
		Collections.sort(alphaList, (a,b) -> {
		   String parts1[] = a.trim().split("\\s+", 2);
		   String parts2[] = b.trim().split("\\s+", 2);
		   
		   if (parts1[1].compareToIgnoreCase(parts2[1]) == 0) {
		       // if log same (exclude id), then compare ID
		       return parts1[0].compareToIgnoreCase(parts2[0]);
		   } else {
		       return parts1[1].compareToIgnoreCase(parts2[1]);
		   }
		});
		
		list.addAll(alphaList);
		list.addAll(digitList);
		return list;
	}
	// METHOD SIGNATURE ENDS
}

File: NO1/untitled.cs
using System;
using System.Collections.Generic;
using System.Text.RegularExpressions;
using System.Linq;
// IMPORT LIBRARY PACKAGES NEEDED BY YOUR PROGRAM
// SOME CLASSES WITHIN A PACKAGE MAY BE RESTRICTED
// DEFINE ANY CLASS AND METHOD NEEDED
// CLASS BEGINS, THIS CLASS IS REQUIRED
public class Program
{
	// METHOD SIGNATURE BEGINS, THIS METHOD IS REQUIRED
	public List<String> retrieveMostFrequentlyUsedWords(String literatureText, 
	                                               List<String> wordsToExclude)
	{
		List<String> results = new List<String>();
		int max = 0;
		
		if (literatureText == null || literatureText.Length == 0)
		    return results;
		    
		string pattern = @"[a-zA-Z]+";
		string input = literatureText.ToLower();
		
		RegexOptions options = RegexOptions.Singleline;
		
		Dictionary<string, int> dict = new Dictionary<string, int>();
		
		foreach (Match m in Regex.Matches(input, pattern, options)) {
		    string str = m.Value;
		    
		    // if match word listed in exclude, skip
		    if (wordsToExclude != null && wordsToExclude.Contains(str, StringComparer.OrdinalIgnoreCase))
		        continue;
		        
		    if (dict.ContainsKey(str)) {
		        dict[str] += 1;
		    } else {
		        dict.Add(str, 1);
		    }
		    
		    if (max < dict[str])
		        max = dict[str];
		}
		
		foreach (KeyValuePair<string, int> pair in dict) {
		    if (pair.Value == max) {
		        results.Add(pair.Key);
		    }
		}
		
		return results;
		
	}
	// METHOD SIGNATURE ENDS
}